### DISCUSSION TITLE
Validation callback only called once if no value returned by it

### DISCUSSION THREAD

**Discussion by BloodyRain2k** [alejandroautalan/pygubu-designer#210] [Q&A] [answered]
So I was working on another PR for the designer's UI and I needed to get a callback for some of the code template properties.

Being completely new to Tk and barely having started half a year ago with Python, I just went with the "validate" callbacks.

But since I don't know the inner workings of either Tk nor PyGubu, I found out that if a "validate" callback doesn't return *any* value, it is never called a 2nd time.

Is that on purpose?

**Comment by alejandroautalan** [comment, marked as answer]
Hello @BloodyRain2k,

> But since I don't know the inner workings of either Tk nor PyGubu, I found out that if a "validate" callback doesn't return any value, it is never called a 2nd time.
> Is that on purpose?

Yes, that is the default behavior. If you don't return a boolean value the validation is disabled. According to tk [documentation](https://www.tcl.tk/man/tcl8.6/TkCmd/ttk_entry.html#M34): 

_The Tk themed entry widget only disables validation if one of the validation scripts raises an error, or if -validatecommand does not return a valid boolean value. (Thus, it is not necessary to re-enable validation after modifying the entry value in a validation script)._

You have some validation examples here:

https://github.com/alejandroautalan/pygubu-designer/blob/7153c7ef6021ff3889c3898f40d9667300dbd81d/examples/user_input/userinputapp.py#L82-L89

And here:

https://github.com/alejandroautalan/pygubu-designer/blob/7153c7ef6021ff3889c3898f40d9667300dbd81d/examples/7guis/02_temperature_converter/tempconvapp.py#L39-L51

Regards
Alejandro A.
